### ISSUE TITLE
SSL issues

### ISSUE THREAD

**Issue by outcatcher** [opentelekomcloud-infra/csm-sandbox#35]
Changes done in #34 include SSL setup for nginx.

www.ssllabs.com find following problems (https://www.ssllabs.com/ssltest/analyze.html?d=csm.outcatcher.com):

```
# TLS 1.2                                                 (suites in server-preferred order)
--
TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384 (0xc030)                                                                                    ECDH x25519 (eq. 3072 bits RSA)   FS | 256
TLS_DHE_RSA_WITH_AES_256_GCM_SHA384 (0x9f)                                                                                                                              DH 1024 bits   FS                                                                                            WEAK | 256
TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256 (0xcca8)                                                                                    ECDH x25519 (eq. 3072 bits RSA)   FS | 256
TLS_DHE_RSA_WITH_CHACHA20_POLY1305_SHA256 (0xccaa)                                                                                                                              DH 1024 bits   FS                                                                                            WEAK | 256
TLS_DHE_RSA_WITH_AES_256_CCM_8 (0xc0a3)                                                                                                                              DH 1024 bits   FS                                                                                            WEAK | 256
TLS_DHE_RSA_WITH_AES_256_CCM (0xc09f)                                                                                                                              DH 1024 bits   FS                                                                                            WEAK | 256
TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256 (0xc02f)                                                                                    ECDH x25519 (eq. 3072 bits RSA)   FS | 128
TLS_DHE_RSA_WITH_AES_128_GCM_SHA256 (0x9e)                                                                                                                              DH 1024 bits   FS                                                                                            WEAK | 128
TLS_DHE_RSA_WITH_AES_128_CCM_8 (0xc0a2)                                                                                                                              DH 1024 bits   FS                                                                                            WEAK | 128
TLS_DHE_RSA_WITH_AES_128_CCM (0xc09e)                                                                                                                              DH 1024 bits   FS                                                                                            WEAK | 128
TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384 (0xc028)                                                               ECDH x25519 (eq. 3072 bits RSA)   FS                                                       WEAK | 256
TLS_DHE_RSA_WITH_AES_256_CBC_SHA256 (0x6b)                                                                                                         DH 1024 bits   FS                                                                   WEAK | 256
TLS_ECDHE_RSA_WITH_CAMELLIA_256_CBC_SHA384 (0xc077)                                                               ECDH x25519 (eq. 3072 bits RSA)   FS                                                       WEAK | 256
TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA256 (0xc4)                                                                                                         DH 1024 bits   FS                                                                   WEAK | 256
TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256 (0xc027)                                                               ECDH x25519 (eq. 3072 bits RSA)   FS                                                       WEAK | 128
TLS_DHE_RSA_WITH_AES_128_CBC_SHA256 (0x67)                                                                                                         DH 1024 bits   FS                                                                   WEAK | 128
TLS_ECDHE_RSA_WITH_CAMELLIA_128_CBC_SHA256 (0xc076)                                                               ECDH x25519 (eq. 3072 bits RSA)   FS                                                       WEAK | 128
TLS_DHE_RSA_WITH_CAMELLIA_128_CBC_SHA256 (0xbe)                                                                                                         DH 1024 bits   FS                                                                   WEAK | 128
TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA (0xc014)                                                               ECDH x25519 (eq. 3072 bits RSA)   FS                                                       WEAK | 256
TLS_DHE_RSA_WITH_AES_256_CBC_SHA (0x39)                                                                                                         DH 1024 bits   FS                                                                   WEAK | 256
TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA (0x88)                                                                                                         DH 1024 bits   FS                                                                   WEAK | 256
TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA (0xc013)                                                               ECDH x25519 (eq. 3072 bits RSA)   FS                                                       WEAK | 128
TLS_DHE_RSA_WITH_AES_128_CBC_SHA (0x33)                                                                                                         DH 1024 bits   FS                                                                   WEAK | 128
TLS_DHE_RSA_WITH_CAMELLIA_128_CBC_SHA (0x45)                                                                                                         DH 1024 bits   FS                                                                   WEAK | 128
TLS_RSA_WITH_AES_256_GCM_SHA384 (0x9d)                                                                                       WEAK | 256
TLS_RSA_WITH_AES_256_CCM_8 (0xc0a1)                                                                                       WEAK | 256
TLS_RSA_WITH_AES_256_CCM (0xc09d)                                                                                       WEAK | 256
TLS_RSA_WITH_AES_128_GCM_SHA256 (0x9c)                                                                                       WEAK | 128
TLS_RSA_WITH_AES_128_CCM_8 (0xc0a0)                                                                                       WEAK | 128
TLS_RSA_WITH_AES_128_CCM (0xc09c)                                                                                       WEAK | 128
TLS_RSA_WITH_AES_256_CBC_SHA256 (0x3d)                                                                                       WEAK | 256
TLS_RSA_WITH_CAMELLIA_256_CBC_SHA256 (0xc0)                                                                                       WEAK | 256
TLS_RSA_WITH_AES_128_CBC_SHA256 (0x3c)                                                                                       WEAK | 128
TLS_RSA_WITH_CAMELLIA_128_CBC_SHA256 (0xba)                                                                                       WEAK | 128
TLS_RSA_WITH_AES_256_CBC_SHA (0x35)                                                                                       WEAK | 256
TLS_RSA_WITH_CAMELLIA_256_CBC_SHA (0x84)                                                                                       WEAK | 256
TLS_RSA_WITH_AES_128_CBC_SHA (0x2f)                                                                                       WEAK | 128
TLS_RSA_WITH_CAMELLIA_128_CBC_SHA (0x41)                                                                                       WEAK | 128
# TLS 1.1                                                 (suites in server-preferred order)
TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA (0xc014)                                                               ECDH x25519 (eq. 3072 bits RSA)   FS                                                       WEAK | 256
TLS_DHE_RSA_WITH_AES_256_CBC_SHA (0x39)                                                                                                         DH 1024 bits   FS                                                                   WEAK | 256
TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA (0x88)                                                                                                         DH 1024 bits   FS                                                                   WEAK | 256
TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA (0xc013)                                                               ECDH x25519 (eq. 3072 bits RSA)   FS                                                       WEAK | 128
TLS_DHE_RSA_WITH_AES_128_CBC_SHA (0x33)                                                                                                         DH 1024 bits   FS                                                                   WEAK | 128
TLS_DHE_RSA_WITH_CAMELLIA_128_CBC_SHA (0x45)                                                                                                         DH 1024 bits   FS                                                                   WEAK | 128
TLS_RSA_WITH_AES_256_CBC_SHA (0x35)                                                                                       WEAK | 256
TLS_RSA_WITH_CAMELLIA_256_CBC_SHA (0x84)                                                                                       WEAK | 256
TLS_RSA_WITH_AES_128_CBC_SHA (0x2f)                                                                                       WEAK | 128
TLS_RSA_WITH_CAMELLIA_128_CBC_SHA (0x41)                                                                                       WEAK | 128
# TLS 1.0                                                 (suites in server-preferred order)
TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA (0xc014)                                                               ECDH x25519 (eq. 3072 bits RSA)   FS                                                       WEAK | 256
TLS_DHE_RSA_WITH_AES_256_CBC_SHA (0x39)                                                                                                         DH 1024 bits   FS                                                                   WEAK | 256
TLS_DHE_RSA_WITH_CAMELLIA_256_CBC_SHA (0x88)                                                                                                         DH 1024 bits   FS                                                                   WEAK | 256
TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA (0xc013)                                                               ECDH x25519 (eq. 3072 bits RSA)   FS                                                       WEAK | 128
TLS_DHE_RSA_WITH_AES_128_CBC_SHA (0x33)                                                                                                         DH 1024 bits   FS                                                                   WEAK | 128
TLS_DHE_RSA_WITH_CAMELLIA_128_CBC_SHA (0x45)                                                                                                         DH 1024 bits   FS                                                                   WEAK | 128
TLS_RSA_WITH_AES_256_CBC_SHA (0x35)                                                                                       WEAK | 256
TLS_RSA_WITH_CAMELLIA_256_CBC_SHA (0x84)                                                                                       WEAK | 256
TLS_RSA_WITH_AES_128_CBC_SHA (0x2f)                                                                                       WEAK | 128
TLS_RSA_WITH_CAMELLIA_128_CBC_SHA (0x41)                                                                                       WEAK | 128

```

Recommended fix: https://weakdh.org/sysadmin.html

**Comment by anton-sidelnikov** [comment]
fix: https://weakdh.org/sysadmin.html

**Comment by outcatcher** [comment]
Closed by #36
